src/psyclone/errors.py

```diff
@@ -32,51 +32,56 @@
 # POSSIBILITY OF SUCH DAMAGE.
 # -----------------------------------------------------------------------------
 # Authors R. W. Ford, A. R. Porter and S. Siso, STFC Daresbury Lab
-#         I. Kavcic, Met Office
+#         I. Kavcic and A. J. Voysey, Met Office
 #         J. Henrichs, Bureau of Meteorology
 # -----------------------------------------------------------------------------
 
 ''' This module provides various error classes used in PSyclone'''
 
-
-class GenerationError(Exception):
-    ''' Provides a PSyclone specific error class for errors found during PSy
-    code generation.
+class PsycloneError(Exception):
+    ''' Provides a PSyclone specific error class as a generic parent class for
+    all Pysclone exceptions.
 
     :param str value: the message associated with the error.
     '''
     def __init__(self, value):
         Exception.__init__(self, value)
-        self.value = "Generation Error: "+value
+        self.value = "Psyclone Error: " + str(value)
+
+    def __repr__(self):
+        return type(self).__name__ + "()"
 
     def __str__(self):
         return str(self.value)
 
 
-class FieldNotFoundError(Exception):
-    ''' Provides a PSyclone-specific error class when a field with the
-    requested property/ies is not found.
+class GenerationError(PsycloneError):
+    ''' Provides a PSyclone specific error class for errors found during PSy
+    code generation.
 
     :param str value: the message associated with the error.
     '''
     def __init__(self, value):
-        Exception.__init__(self, value)
-        self.value = "Field not found error: "+value
+        PsycloneError.__init__(self, value)
+        self.value = "Generation Error: "+str(value)
 
-    def __str__(self):
-        return str(self.value)
+class FieldNotFoundError(PsycloneError):
+    ''' Provides a PSyclone-specific error class when a field with the
+    requested property/ies is not found.
 
+    :param str value: the message associated with the error.
+    '''
+    def __init__(self, value):
+        PsycloneError.__init__(self, value)
+        self.value = "Field not found error: "+str(value)
 
-class InternalError(Exception):
+class InternalError(PsycloneError):
     '''
     PSyclone-specific exception for use when an internal error occurs (i.e.
     something that 'should not happen').
 
     :param str value: the message associated with the error.
     '''
     def __init__(self, value):
-        Exception.__init__(self, value)
-        self.value = "PSyclone internal error: "+value
-
-    def __str__(self):
-        return str(self.value)
+        PsycloneError.__init__(self, value)
+        self.value = "PSyclone internal error: "+str(value)
```

Comment:
Not your fault but, since we're here, please could you add an `__all__` at the bottom of this file and list the classes it contains:

    # For Sphinx AutoAPI documentation generation
    __all__ = ["PSycloneError", etc. etc.]